kit/config/decode.go

```diff
@@ -0,0 +1,178 @@
+package config
```

Comment:
I had created dapr/dapr/issues/2820 to discuss a `kit` repo for reuse across the various other Go-based Dapr repos. I named the package in this PR `kit` because I eventually saw it moving to that, but I agree that `internal` is the logic choice here as the initial step and decouples it entirely from that discussion.